validator/client/validator.go

```diff
@@ -120,6 +112,57 @@ func (v *validator) NextSlot() <-chan uint64 {
 	return v.ticker.C()
 }
 
+// LogValidatorGainsAndLosses logs important metrics related to this validator client's
+// responsibilities throughout the beacon chain's lifecycle. It logs absolute accrued rewards
+// and penalties over time, percentage gain/loss, and gives the end user a better idea
+// of how the validator performs with respect to the rest.
+func (v *validator) LogValidatorGainsAndLosses(ctx context.Context, slot uint64) error {
+	if slot%params.BeaconConfig().SlotsPerEpoch != 0 {
+		// Do nothing if we are not at the start of a new epoch.
+		return nil
+	}
+	epoch := slot / params.BeaconConfig().SlotsPerEpoch
+	if epoch == params.BeaconConfig().GenesisEpoch {
+		v.prevBalance = params.BeaconConfig().MaxDepositAmount
+	}
+	req := &pb.ValidatorPerformanceRequest{
+		Slot:      slot,
+		PublicKey: v.key.PublicKey.Marshal(),
+	}
+	resp, err := v.validatorClient.ValidatorPerformance(ctx, req)
+	if err != nil {
+		return err
+	}
+	newBalance := float64(resp.Balance) / float64(params.BeaconConfig().GweiPerEth)
+	log.WithFields(logrus.Fields{
+		"slot":  slot - params.BeaconConfig().GenesisSlot,
+		"epoch": (slot / params.BeaconConfig().SlotsPerEpoch) - params.BeaconConfig().GenesisEpoch,
+	}).Info("Start of a new epoch!")
+	log.WithFields(logrus.Fields{
+		"totalValidators":     resp.TotalValidators,
+		"numActiveValidators": resp.TotalActiveValidators,
+	}).Infof("Validator registry information")
+	log.Info("Generating validator performance report from the previous epoch...")
+	log.WithFields(logrus.Fields{
+		"ethBalance": newBalance,
+	}).Info("New validator balance")
```

Comment:
Can you combine the status update into a single log statement? 

I'm just thinking about log entries in some searchable feed. This method prints up to 7 distinct INFO when it could be 1 INFO with all of the fields & information.